Resurrect rush

s-mage opened the pull request:
1. Refactor the code
2. Retab
3. Make all specs passed
4. Add some heuristics and aliases

And I really want to maintain it. If noone still doing it, may I?

s-mage committed:
Some cleanup and retab

schfkt committed:
Ignore .swp files

s-mage committed:
Retab, hack to run bash commands without pain

s-mage committed:
Some bash-like aliases, xdg-open and locate

s-mage committed:
Add usage comment for open

s-mage committed:
Add open_with function to open files with any application you like

s-mage committed:
Add multiline input support

s-mage committed:
Add pager to printing results, colorize less

s-mage committed:
Fix autocomplete, make it work with constants

s-mage committed:
Clean code a bit

s-mage committed:
Specs now running. But not passed

s-mage committed:
Fix access inconsistence

s-mage committed:
Fix access specs

s-mage committed:
All specs passed

s-mage committed:
Now you can tune your prompt

Sija commented on tags:
```diff
@@ -0,0 +1,312 @@
+!_TAG_FILE_FORMAT	2	/extended format; --format=1 will not append ;" to lines/
```
Not needed i suppose :)

s-mage replied:
Not needed line or all file? It's ctags file and it 'just works' for me. We can remove it and add to gitignore, if noone use ctags extept me.

Sija replied:
Whole file, i'm for .gitignore-ing it as it's a personal artefact.

s-mage replied:
ok, I did it. Thank you for attentiveness.

Sija replied:
no problemo! thx for will to maintain Rush, i love it too! :)

On 23 Mar 2014, at 17:06, Sergey Smagin notifications@github.com wrote:

> In tags:
> 
> > @@ -0,0 +1,312 @@
> > +!_TAG_FILE_FORMAT  2   /extended format; --format=1 will not append ;" to lines/
> 
> ok, I did it. Thank you for attentiveness.
> 
> —
> Reply to this email directly or view it on GitHub.

s-mage committed:
Remove ctags file and add it go .gitignore

s-mage committed:
More accurate work with bash commands

s-mage committed:
Add travis c support

s-mage committed:
Add rake to dependencies

schfkt committed:
Update gemspec. Fix travis config

schfkt committed:
Add travis badge

schfkt committed:
Fix badge

s-mage committed:
Remove jruby from travis

s-mage committed:
Add rubinius to travis

s-mage committed:
Fix spec for rubinius

s-mage committed:
Replace readline with coolline. Add syntax highlight

s-mage committed:
Remove swap file

schfkt committed:
Fix gemspec

s-mage committed:
Small improvements

najamelan commented:
Hi,

I have yet to use rush, but it looks like an interesting project. If it suits my needs, I'll happily try to contribute to the codebase. S-mage, I propose we consider your fork to be the new project home. For now, I am porting the rush handbook that was hosted on heroku but now only available on the internet archive to the wiki of your fork. That way we can centralize a bit the existing documentation. I hope that's ok with you?
Thanks for maintaining this.

s-mage commented:
Oh, it will be really good! Thank you for that!

On 2 May 2014 15:32, Naja Melan notifications@github.com wrote:

> Hi,
> 
> I have yet to use rush, but it looks like an interesting project. If it
> suits my needs, I'll happily try to contribute to the codebase. S-mage, I
> propose we consider your fork to be the new project home. For now, I am
> porting the rush handbook that was hosted on heroku but now only available
> on the internet archive to the wiki of your fork. That way we can
> centralize a bit the existing documentation. I hope that's ok with you?
> Thanks for maintaining this.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/adamwiggins/rush/pull/19#issuecomment-42020191
> .

najamelan commented:
Ok, 
it's done. May I ask you some favors? When committing:
- favor small self-contained commits
- always separate code formatting/whitespace changes from actual changes to the code
- write a bit more justification in comments/commit messages as to why and how you make changes. Even one or two phrases make it much more easy to understand the motivation and possibly the consequences of a change. (possibly keep a changelog for anything but refactoring)

That makes it easier for other people to understand what's happening to the code. Consider that the commit history is your blog. People read it. :wink: 

ps: i will do a pull requests. I will not be offended if you decline them. They're just propositions.

Commit:
Add installation instructions to the README

On copying the documentation from heroku I came accross installation instructions that seem to have gotten lost when the project came to github. I felt it would be useful to reintegrate them in the README. I don't know if there was a motivation for not bringing them in from heroku.

Commit:
add uninstall instructions also

I'm not a veteran ruby expert, I don't know if this suffices.

Commit:
Added ruby-dev to requirements in readme

For the gem install --dev option

Commit:
Add documentation and .gem file to .gitignore

This adds 
*.gem
doc/
.yardoc/

Commit:
Update website and meta in the readme

There's not much point for outdated information

s-mage committed:
Merge pull request #2 from najamelan/hotfix/ignoreDocs

Add documentation and .gem file to .gitignore

s-mage committed:
Merge pull request #3 from najamelan/hotfix/outdatedReadme

Update website and meta in the readme

s-mage committed:
Improve work with requirements. Remove circular dependency and some "require rubygems"

s-mage committed:
Add development group to Gemfile

Commit:
Deprecated function in rspec2: stub!

Replaced by stub

Commit:
Use rspec2 instead of legacy rspec

All tests pass, but with deprecated methods.

Commit:
Deprecated methods in rspec2: mock

Replaced by double
All tests run cleanly

Commit:
Merge branch 'master' into hotfix/readmeInstallInstructions

* master:
  Add development group to Gemfile
  Improve work with requirements. Remove circular dependency and some "require rubygems"
  Update website and meta in the readme
  Add documentation and .gem file to .gitignore

Commit:
Add .bundle to .gitignore

As per http://bundler.io/v1.3/rationale.htm this is automated configuration files which should not be under version control.

Commit:
Update the bundle dependencies to include rspec 2

This update was done running bundle update to update all dependencies. All tests pass.

Gemfile.lock add's hard dependency versions to this gem. 
It was forcing rspec down to 1.2.6.

This reference (http://bundler.io/v1.3/rationale.html) seems to say that you should check it into versioning, but then it would be good to explain somewhere in development docs that you need to regenerate Gemfile.lock with 'bundle update' before pushing and that you should test if things work with bundle install. I just completely bypassed bundle causing this version conflict.

See also this, which seems to claim it's a bad idea to put Gemfile.lock in version control: 
http://yehudakatz.com/2010/12/16/clarifying-the-roles-of-the-gemspec-and-gemfile/

A good explanation can also be found in "man bundle install", especially the section on deployment mode.

Commit:
Try to work around travis bug travis-ci/travis-ci#2220

Not sure if it'll work but it's worth to try

s-mage committed:
Merge pull request #4 from najamelan/feature/rspec2

Update rspec to rspec2

s-mage committed:
Merge pull request #1 from najamelan/hotfix/readmeInstallInstructions

Add installation instructions to the README

Commit:
remove a double definition of exceptions

These exceptions are already defined in exceptions.rb, so I think they shouldn't be defined in entry.rb.

I didn't try to manually figure out if this might break anything, but all tests pass.

Commit:
Do not unshift ourselves on the $LOADPATH since we now use require_relative

no need to touch it and bad practice anyways

Commit:
Installation instruction for development

REVIEW: I just wrote this as well as I could, but I'm no ruby expert. Needs verifying.

s-mage committed:
Merge pull request #6 from najamelan/hotfix/cleanup

Remove some perl-style requirements.

Commit:
Add vendor/bundle to gitignore

Bundler installs dependencies to this subfolder when calling "bundle install". They should not be tracked by git.

Commit:
Simplify dev install instructions by using rake install.

I'm still not sure about the many different ways of doing stuff in ruby gems. Note that sudo gem install had --development option. I'm not sure what rake install exactly does, but it is briefer.

s-mage committed:
Merge pull request #5 from najamelan/hotfix/devInstallInstructions

Installation instruction for development

Commit:
remove a backup file

Commit:
Create Entry::chown and Entry::chown_R

Rush didn't have a method for changing ownership of files and directories. This adds Entry::chown which takes 3 parameters:
user, group and options

It is a simple frontend for FileUtils.chown and chown_R, to integrate their functionality with the Rush syntax.

Currently not very tested. The problem is that tests for this will have to run as sudo usually, because unprivileged you can't do much with chmod. Also, I currently didn't have a remote setup which would be practical for testing, so remote testing has not been done at all.

Towards the future I would like to add more sys admin functionality to Rush, so it's not only an interface to filesystem and bash, but also to user management. Obviously a growing number of features would require root access to be used, and thus also to be tested. I will think about a way to create an optional set of tests that will be run only if the specs are being run as root. That way, we can have our tests, but peopl who can't be bothered to run it as root can still run all the other tests.

s-mage committed:
Merge pull request #7 from najamelan/feature/chown

Create Entry::chown and Entry::chown_R

s-mage committed:
Refactor pull request #7

s-mage committed:
Add owner accessor to file

s-mage committed:
Merge pull request #8 from najamelan/hotfix/removeBackupFile

remove a backup file

s-mage committed:
Fix crashing on some cases of autocomplete

s-mage committed:
[refs #10] Remove mongrel, bring connection via net/ssh. Still no rush magic, but I am working on it.

s-mage committed:
Remove .idea

s-mage committed:
Dir and config specs moved to expect syntax

s-mage committed:
Search results and string ext specs moved to expect syntax

s-mage committed:
Add completion for executables

s-mage committed:
[refs #9] Rewrite autocomplete system

s-mage committed:
Output should be pretty again

s-mage committed:
Run executables without pain: my_file.firefox will open firefox

s-mage committed:
Update dependencies

s-mage committed:
[refs #9] Tests for completion

s-mage committed:
[refs #10] Use sshfs for remote connections

s-mage committed:
Version bump to 0.7.0

s-mage committed:
Update readme; release gem rush2

s-mage committed:
Improve autocompletion with coolline settings

s-mage committed:
Version bump to 0.7.1

s-mage committed:
Split shell initialization into several commands

s-mage committed:
Update dependencies

s-mage committed:
Move more specs to new format

s-mage committed:
Moving specs to new syntax -- box_spec

s-mage committed:
Update dependencies; rewrite some tests

s-mage committed:
[refs #13] Hash of keys in open_with and output_of

s-mage committed:
[refs #13] Support -k and --key

s-mage committed:
[refs #13] Update documentation; fix open_command

s-mage committed:
Add edit command

s-mage committed:
Fix process spec and shell spec

s-mage committed:
Fix file spec

s-mage committed:
Fix process spec

s-mage committed:
Remove remote and spec for it. Noone use it

s-mage committed:
Fix entry spec

s-mage committed:
All specs passed!

s-mage committed:
No deprecation warnings anymore

s-mage committed:
Update dependencies; bump version

s-mage committed:
Small code style improvements

razum2um committed:
dry path manipulation & fix travis-ci

razum2um committed:
test on 2.2

razum2um committed:
support Dir#size on OSX

razum2um committed:
allow sudo in travis

razum2um committed:
rbx support

razum2um committed:
support 1.9.3

razum2um committed:
support osx subprocess test

s-mage committed:
Merge pull request #14 from razum2um/master

Rush::Path, OSX specs, travis settings done right

s-mage committed:
Update dependencies

s-mage committed:
Merge branch 'master' of github.com:s-mage/rush

jikkujose commented:
Is this remaining unmerged because of the lack of a maintainer?

s-mage commented:
@JikkuJose I guess yes. You can use my version (I am humble, I know).

s-mage commented:
@rickhull because it never worked properly :)
Sorry, I just gave up with it. You can provide an idea to implement it.

rickhull commented:
well, i'm the new maintainer of net-ssh-shell too, but only because I have an interest, not expertise.

SIGH

oh well, I'll try to step up

jikkujose commented:
I emailed @adamwiggins but haven't got any reply! His last activity seems to be in 2015

s-mage committed:
Add optional environment variables to commands

s-mage committed:
Document changes about env option

s-mage committed:
better ls implementation

s-mage committed:
Print error backtrace only by request (wtf method)

s-mage committed:
Better paths completion

s-mage committed:
update travis config

s-mage committed:
update dependencies; remove redundant method

s-mage committed:
Remove redundant method

s-mage committed:
change badge to drone.io one

s-mage committed:
update drone badge

s-mage committed:
typo

s-mage committed:
Add basic support to execute strings with executables

s-mage committed:
Add very basic support of pipes

s-mage committed:
Dir contents is now includes hidden files

s-mage committed:
Quote paths before opening apps

s-mage committed:
Make locate case-insensitive

s-mage committed:
Fix tests; refactor open_command

s-mage committed:
Bump version

s-mage committed:
Generate gemspec

s-mage committed:
Update dependencies

Commit:
Add functionality for hard and symlinking files and directories

Commit:
rename slink to symlink

s-mage committed:
Merge pull request #20 from najamelan/feature/symlink

Add functionality for hard and symlinking files and directories

s-mage committed:
Dont track ruby version

s-mage committed:
Version bump to 0.10.0

s-mage committed:
Updated dependencies; minor refactoring; version bump

s-mage committed:
Updated gemspec
